morf-core/src/test/java/org/alfasoftware/morf/jdbc/TestResultSetIterator.java

```diff
@@ -208,6 +214,94 @@ public void testBuildWithEmptyColumnOrdering() throws Exception {
   }
 
 
+  /**
+   * Tests building a ResultSetIterator with a query that produces a result set using the configured fetch size.
+   * @throws Exception
+   */
+  @Test
+  public void testQueryWithResultSetAndFetchSize() throws Exception {
+    // Given
+    Table table = buildTable();
+    String query = "select column from table";
+    ResultSet resultSet = mock(ResultSet.class);
+    given(statement.executeQuery(query)).willReturn(resultSet);
+    given(resultSet.findColumn("Column")).willReturn(1);
+    given(resultSet.next()).willReturn(true).willReturn(true).willReturn(false);
+    given(connectionResources.getFetchSizeForBulkSelects()).willReturn(1000);
+
+    // When
+    @SuppressWarnings("resource") /* Resources are closed in ResultSetIterator.close()
+                                     automatically when caller attempts to advance past
+                                     the last row in the result set. */
+        ResultSetIterator resultSetIterator = new ResultSetIterator(table, query, connection, Optional.of(connectionResources), sqlDialect);
+
+    // Then
+    assertTrue(resultSetIterator.hasNext());
+
+    resultSetIterator.next();

```

Comment:
Why do we need to test iteration until an exception is thrown? It's not necessary for testing whether the fetch size was set, so I think these unit tests should be a lot simpler.